### PR TITLE
Add @types files to node-rules

### PR DESCRIPTION
**Pull request by xiaoxizi2303**
Add @types files to node-rules

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/node-rules/node-rules-tests.ts
1. types/node-rules/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 653e2e9de2b0099f10e9660fefe00aaa89122381
</p>

File: types/node-rules/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+     * The consequence to be executed if the condition is met.
+     * @param R The rule engine instance.
+     * @param fact The fact being evaluated.
+     */
+    consequence: (R: FlowControlAPI, fact: Fact) => void;
+
+    /**
+     * Optional parameter to specify the priority of the rule.
+     */
+    priority?: number;
```
Comment (jakebailey) [inline types/node-rules/index.d.ts:37]: I highly suspect that this should all `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

**Review by Renegade334 — COMMENTED** [review]
This package [already contains type definitions](https://github.com/mithunsatheesh/node-rules/blob/main/dist/index.d.ts)?

**Review by jakebailey — CHANGES_REQUESTED** [review]
Ah, it sure does. This PR is not needed.

**Comment by xiaoxizi2303** [comment]
> This package [already contains type definitions](https://github.com/mithunsatheesh/node-rules/blob/main/dist/index.d.ts)?

Is this the latest version supported?

**Comment by jakebailey** [comment]
It doesn't get more latest than latest: https://unpkg.com/browse/node-rules@latest/dist/index.d.ts

**Comment by jakebailey** [comment]
I don't understand what you mean. If you're saying that their API does not have enough exported, you should report that to them.

Even if you have the package on DT, the real package's types always take precedence over `@types`.

**Comment by xiaoxizi2303** [comment]
> I don't understand what you mean. If you're saying that their API does not have enough exported, you should report that to them.
> 
> Even if you have the package on DT, the real package's types always take precedence over `@types`.

You may be misunderstanding me, but I want to ask how to use the Rule, Fact, and API types in the declaration file that comes with your library, because when I use these types, the error message is because these interfaces are not exported

**Comment by jakebailey** [comment]
You don't; you have to ask the authors of the library to publish different types, or copy and paste them yourself. Or fork their library.

(It's not my library, I'm just a maintainer of DefinitelyTyped.)

**Comment by jakebailey** [comment]
It's also possible to use things like `ReturnType`, `Parameters`, etc, to extract out their types via their public API. But, changing DT unfortunately is not the solution here.

**Comment by xiaoxizi2303** [comment]
Ok, I see. Thanks for the answer.

**Pull request closed without merging**